Lib/Environment.hpp

```diff
@@ -93,11 +92,24 @@ class Environment
   /** set to true when coloring is used for symbol elimination or interpolation */
   bool colorUsed;
 
+  /**
+   * A global way of accessing "the problem vampire is working on", maily for checking its properties.
+   * Note that if in some special cases there is more than one Problem instance used at one time moment,
+   * one should know which is the main one and that one should be set/reset here.
+   *
+   * (In an ideal world, there would be no need for this function, as the correct Problem object would
+   * be explicitly passed to all the functions interested in knowning...)
+   */
+  Kernel::Problem* getMainProblem() { return _problem; }
```

Comment:
I guess we have to have the pointer indirection as we can't copy/move a `Problem`?

Reply:
Also, do we want the getter/setter or a bare object? I know this is considered best practice in some languages. C++ I'm not sure.